erpnext/stock/doctype/putaway_rule/putaway_rule.py

```diff
@@ -83,7 +83,7 @@ def apply_putaway_rule(doctype, items, company, sync=None, purpose=None):
 		if isinstance(item, dict):
 			item = frappe._dict(item)
 
-		source_warehouse = item.get("s_warehouse")
+		source_warehouse = item.get("s_warehouse") or item.get("t_warehouse")
```

Comment:
Lets keep it as it is, if no source warehouse let it be blank. this variable is passed on to `get_ordered_putaway_rules` that excludes rules having this source warehouse. That means we want to include the target warehouse so we make sure that doesnt go to `get_ordered_putaway_rules`.
```suggestion
		source_warehouse = item.get("s_warehouse")
```